### DISCUSSION TITLE
苍何大佬可以分享下如何高效学习编程吗？

### DISCUSSION THREAD

**Discussion by hissenss** [freestylefly/CodeCanvas#23] [General]
每次总是记不住，看视频也记不住，然后真正自己写的时候又忘记了。

**Comment by freestylefly** [comment]
你好好，你的问题相信也是大部分同学的问题，包括以前的我也面临同样的问题，这里分享下自己的学习经验，供你做下参考：

第一，学习知识的时候，确定好目标。比如学习 spring 的视频，你的目标到底是什么？是要掌握这门知识吗？我觉得有效的目标应该不止于此，掌握知识只是目的而不是结果，你应该要以量化的结果作为自己的目标，比如，学完黑马的 springboot 视频，我的目标就是要利用 springboot 自己独立完成一个项目，而不是一味的去研究底层技术，当有这个目标后， 对于一些原理类的可以暂时跳过，等过完一遍，完成独立做项目后，再来深挖原理。

第二，我推荐的比较好的学习方式是「教中学」，在学习知识的时候，怎么样衡量我们对知识是否理解透彻掌握呢？那就是把这些知识用自己的话给别人讲一遍，用教别人知识来鞭策自己更好的学习，如果别人能听懂你讲的知识，表明理解到位了。可以用写博客的方式或者在星球输出打卡，都是一种好的方式。

第三，学习时极度专注，要学就学个踏实，要玩就玩个痛快。建议可用番茄专注，每 30 分钟一个专注，尽量不受外界干扰。

第四，能获得有效的反馈，有效的反馈可以来自外部也可以是内部，比如学完 spring 的 aop 后，你可以试着用自己的话总结什么是面向切面编程，到底能解决什么问题，星球或者输出博客后，如果能有人给你反馈，帮助是很大的。

第五，知识在于精而不是多，其实对于做项目而言，黑马机构讲的那些视频，做项目完全没问题，之所以不会是还没形成自己的知识，建议可以再带着目标再来复习一次，并按照我总结的方法试着输出获得反馈。

总结一下就是这 5 点：
1、有定义明确的目标
2、在「教中学」
3、学习时极度专注，要学就学个踏实，要玩就玩个痛快
4、能获得有效的反馈
5、知识在于精而不是多

**Comment by hissenss** [comment]
谢谢
